server/datastore/mysql/hosts.go

```diff
@@ -308,7 +308,7 @@ func (d *Datastore) Host(id uint) (*fleet.Host, error) {
 
 func (d *Datastore) amountEnrolledHosts() (int, error) {
 	var amount int
-	err := d.db.Get(&amount, `SELECT count(*) FROM hosts`)
+	err := d.writer.Get(&amount, `SELECT count(*) FROM hosts`)
```

Comment:
ah, good catch but in this case it's because it is called from `ShouldSendStatistics` which may execute an INSERT (and so uses `writer` itself), so for consistency I used `writer` here, but I'll add a comment to explain that.